crates/protogen/src/lib.rs

```diff
@@ -5,10 +5,56 @@
 //! project. There should be a minimal amount of logic in this crate.
 
 pub mod metastore;
+pub mod rpcsrv;
 pub mod export {
     pub use prost;
 }
 
+pub mod errors {
+    /// Errors related to converting to/from protobuf types.
+    #[derive(thiserror::Error, Debug)]
+    pub enum ProtoConvError {
+        #[error("Field required: {0}")]
+        RequiredField(String),
+
+        #[error("Unknown enum variant for '{0}': {1}")]
+        UnknownEnumVariant(&'static str, i32),
+
+        #[error("Received zero-value enum variant for '{0}'")]
+        ZeroValueEnumVariant(&'static str),
+
+        #[error("Unsupported serialization: {0}")]
+        UnsupportedSerialization(&'static str),
+
+        #[error("Invalid table reference: {0:?}.{1:?}.{2:?}")]
+        InvalidTableReference(String, String, String),
```

Comment:
I did this because debug formatting adds quotation marks, which will be helpful for identifying what's wrong with the reference.

```rust
// Invalid table ref:
(Some("database"), None, "table")

// will print as:
// "database".""."table"
```

IMO more readable than `database..table`. Moreover, complies with how we accept identifiers.